### PR TITLE
fix(developer): check HISTORY.md to get last modified date for keyboard_info and model_info

### PR DESCRIPTION
**Pull request by mcdurdin**
This updates the calculation of the lastModifiedDate property of the .keyboard_info and .model_info file. It now looks for the most recent commit to HISTORY.md, and then falls back to the project folder for .keyboard_info and .model_info.

getLastGitCommitDate now also returns UTC date (it was giving TZ=0 which is affected by daylight saving), and works for a file rather than just for a folder, and a corresponding unit test has been added.

The date strings returned will now always include '.000' msec suffix. I have manually verified that the import into the api.keyman.com database copes with this, and that this change still conforms to the expected schema for .keyboard_info and .model_info files.

Last, but important, it turned out that the unit tests for kmc were not running. This has been resolved in kmc/build.sh.

Fixes: #11793

@keymanapp-test-bot skip

### CODE REVIEW

File: developer/src/kmc/src/util/getLastGitCommitDate.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { execFileSync } from 'child_process';
 
 // RFC3339 pattern, UTC
-export const expectedGitDateFormat = /^\d\d\d\d-\d\d-\d\dT\d\d:\d\d:\d\dZ$/;
+export const expectedGitDateFormat = /^\d\d\d\d-\d\d-\d\dT\d\d:\d\d:\d\d(\.\d\d\d)?Z$/;
```
Comment (darcywong00) [inline developer/src/kmc/src/util/getLastGitCommitDate.ts:4]: Does this impact l.9 on the type of date /time string? Is it an optional milliseconds?

**Review by darcywong00 — APPROVED** [review]
lgtm

File: developer/src/kmc/src/util/getLastGitCommitDate.ts (hunk 1 above)
Comment (mcdurdin) [reply developer/src/kmc/src/util/getLastGitCommitDate.ts:4]: yeah comment should be updated...

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.58-alpha](https://keyman.com/downloads/releases/alpha/18.0.58)

**Pull request merged**